velox/exec/tests/CustomJoinTest.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/dwio/dwrf/test/utils/BatchMaker.h"
+#include "velox/exec/JoinBridge.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+#include <gmock/gmock.h>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+using facebook::velox::test::BatchMaker;
+
+class CustomJoinNode : public core::PlanNode {
+ public:
+  CustomJoinNode(
+      const core::PlanNodeId& id,
+      std::shared_ptr<const core::PlanNode> left,
```

Comment:
We just added PlanNodePtr that can be used to shorten these lengthy codes:

```
PlanNodePtr left,
PlanNodePtr right,
```

Reply:
Fixed